ledger/participant-integration-api/src/test/lib/scala/platform/indexer/ha/TestDBLockStorageBackend.scala

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.ha
+
+import java.sql.{
+  Blob,
+  CallableStatement,
+  Clob,
+  Connection,
+  DatabaseMetaData,
+  NClob,
+  PreparedStatement,
+  SQLWarning,
+  SQLXML,
+  Savepoint,
+  Statement,
+  Struct,
+}
+import java.util.Properties
+import java.util.concurrent.Executor
+import java.{sql, util}
+
+import com.daml.platform.store.backend.DBLockStorageBackend
+
+class TestDBLockStorageBackend extends DBLockStorageBackend {
+
+  private var locks: Map[DBLockStorageBackend.Lock, Set[Connection]] = Map.empty
+
+  override def tryAcquire(
```

Comment:
This is a fair amount of code. It looks ok, but we should be aware that the locking behavior of real databases can behave differently.

Reply:
They can yes, but for what it is important to us that both this and both real databases must confirm to our expectations.
These expectations are encapsulated at StorageBackendTestsDBLock.
And that's why I've added TestDBLockStorageBackendSpec, to have this conformance given for our Ha testing.